_source/_posts/2021-02-22-first-nestjs-application.md

```diff
@@ -0,0 +1,305 @@
+---
+layout: blog_post
+title: "Build Your First NestJS Application"
+author: nickolas-fisher
+by: contractor
+communities: [javascript]
+description: "Learn how to build a basic application using NestJS."
+tags: [nestjs, javascript]
+tweets:
+- "Wanna learn how to build a basic application in NestJS? Check this out!"
+- "Need to get started with NestJS? We've got you covered!"
+- "Get your head around the basics of building applications with NestJS!"
+image: blog/first-nestjs-application/social-image.png
+type: conversion
+---
+
+NestJs is a popular Node.js framework that is built with typescript and makes use of object-oriented programming, functional programming, and functional reactive programming. NestJs boasts that it provides a framework for building scalable server-side applications. 
+
+NestJs integrates nicely with Okta's single sign-on provider. Okta makes securing a web service, such as the one you will build, quick and easy. 
+
+## Create your Okta Application with the CLI
+
+The first thing you will need to do is sign up for an [Okta Developer's account](https://developer.okta.com/signup/) if you haven't already. his account is free forever and allows you to start building applications secured with Okta immediately. Next, you will need to create an application in Okta. To do this you will use the [Okta CLI](https://developer.okta.com/blog/2020/12/10/introducing-okta-cli). If you haven't worked with the CLI yet you can check out the [full documentation here](https://cli.okta.com/).
+
+Start by opening the terminal application of your choice. In the terminal, use the command `okta apps create` and give the application a meaningful name. I named mine `library-api` but you can name yours whatever you like. Select `Single Page App` as the application type. Even though you are creating an API in this tutorial, you will be using Postman as the "application" that will consume the API. This can be even more beneficial if you actually create a single-page application later. You'll be able to use this client ID and issuer in that application.
+
+ Since you are going to use Postman to test, enter `https://oauth.pstmn.io/v1/callback` as your **Login Redirect URL**. Set the **Logout Redirect URI** to `https://oauth.pstmn.io/`. If your organization has multiple authorization servers, you can use `default` for this tutorial. After the CLI runs for a few minutes, it will return your *Issuer* and *Client ID*. Make note of these as you will need them in your application. 
```

Comment:
It doesn't take a few minutes! How dare you! Maybe a minute... and that's WITH creating an okta org...